include/atomics/receiver_cadmium.hpp

```diff
@@ -0,0 +1,162 @@
+#include <cadmium/modeling/message_bag.hpp>
+#include <limits>
+#include <math.h> 
+#include <assert.h>
+#include <memory>
+#include <iomanip>
+#include <iostream>
+#include <fstream>
+#include <string>
+#include <chrono>
+#include <algorithm>
+#include <limits>
+#include <random>
+#include "../data_structures/message.hpp"
+
+using namespace cadmium;
+using namespace std;
+
+/** 
+* Structure definition for input and output message.
+*/
+
+struct receiver_defs{
+    struct output : public out_port<message_t> {
+    };
+    struct input : public in_port<message_t> {
+    };
+};
+
+/** 
+* Receiver class definition.
+*/
+template<typename TIME>
+class Receiver {
+	using defs = receiver_defs; 
+	public:
+		//Parameters to be overwritten when instantiating the atomic model
+		TIME   PREPARATION_TIME;  /**< TIME constant for delay between input and output */ //!<constant for Time delay .
+		
+		/** 
+		* default constructor for Receiver class
+		* Initialize TIME constant for Delay,
+		* Initialize Acknowledgment number to zero at beginning,
+		* and sending state sets to False - Passive phase.
+		*/
+		Receiver() noexcept {
+			PREPARATION_TIME = TIME("00:00:10");
+			state.acknowledgement_number = 0;
+			state.sending = false;
+		}
+		/**
+		* Structure definition for Receiver state and Acknowledgment number
+		*/
+		struct state_type {
+			int acknowledgement_number; /**< Number which send for acknowledgment. */ //!< Acknowledge Number.
+			bool sending; /**< Sending values: True for Active phase and False for Passive Phase. */ //!< State of the receiver.
+		};
+        state_type state;
+        // ports definition
+        using input_ports = std::tuple<typename defs::input>;
+        using output_ports = std::tuple<typename defs::output>;
+        
+		/** 
+		* Receiver class member function for internal transition which change sending state 
+		* from Active (True) to Passive (False) 
+		*/
+        void internal_transition() {
+        	state.sending = false;
+        }
+		/**
+		* Receiver class member function for external transition
+		* This function make sure that only one message is being sent per time unit, if more than one messages in input.
+		* after that it sets sending state to TRUE(Active phase) and acknowledgment number.
+		* @param e TIME type variable
+		* @param mbs make_message_bags type variable
+		* 
+		*/
+        void external_transition(TIME e,
+        		typename make_message_bags<input_ports>::type mbs) {
+        			if(get_messages<typename defs::input>(mbs).size()>1) {
+        				assert(false && "one message per time uniti");
+        			}
+        			for(const auto &x : get_messages<typename defs::input>(mbs)) {
+        				state.acknowledgement_number = static_cast < int > (x.value);
+        				state.sending = true;
+        			}
+                           
+        }
+        /**
+		* confluence transition calls internal_transition and external_transition respectively.
+		* @param e TIME type variable
```

Comment:
Can you add @param[in] instead @param as the professor said

Reply:
I think this is not mandatory thing.
I follow the doxygen manual format and in that also this is not compulsory.